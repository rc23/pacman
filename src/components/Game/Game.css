.game {
  max-width: 500px;
  margin: 0 auto;
  display: flex;
  justify-content: center;
}

.game-active {
  text-align: center;
  padding: 10px;
}

.game-lost {
  text-align: center;
}

.timer {
  color: #666;
  margin-top: 3px;
  margin-left: 3px;
}
