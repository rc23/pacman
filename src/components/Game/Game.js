import React, { useState, useEffect } from "react";
import "./Game.css";

import { connect } from "react-redux";

import { ConnectedBoard } from "../Board/Board";
import { PlayAgain } from "../PlayAgain/PlayAgain";

import { toggleIsGameOver } from "../../actions";

export const Game = ({ ...props }) => {
  const [secondsLeft, setSecondsLeft] = useState(3);
  const [gameStatus, setGameStatus] = useState("active");

  useEffect(() => {
    if (secondsLeft > 0) {
      const timerId = setTimeout(() => setSecondsLeft(secondsLeft - 1), 1000);

      return () => clearTimeout(timerId);
    }
  });

  useEffect(() => {
    if (secondsLeft === 0) {
      setGameStatus("lost");
      props.toggleIsGameOver(true);
    }
  }, [secondsLeft, props]);

  const startNewGame = () => {
    props.startNewGame();
    props.toggleIsGameOver(false);
  };

  return (
    <div className="game">
      <div className="body">
        <div className="timer">Time Remaining: {secondsLeft}</div>
        <div className="game-active">
          {gameStatus === "active" && !props.isGameOver ? (
            <ConnectedBoard />
          ) : null}
        </div>
        <div className="game-lost">
          {gameStatus === "lost" && props.isGameOver ? (
            <PlayAgain onClick={startNewGame} gameStatus={gameStatus} />
          ) : null}
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  isGameOver: state.game.player.isGameOver,
});

const mapDispatchToProps = {
  toggleIsGameOver,
};

export const ConnectedGame = connect(
  mapStateToProps,
  mapDispatchToProps
)(Game);
