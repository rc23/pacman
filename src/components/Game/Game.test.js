import React from "react";
import renderer from "react-test-renderer";

import Game from "./Game";

describe("Game", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Game />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
