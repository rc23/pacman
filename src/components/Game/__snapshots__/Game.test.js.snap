// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders correctly 1`] = `
<div
  className="game"
>
  <div
    className="help"
  >
    Pacman
  </div>
  <div
    className="body"
  >
    <div
      className="left"
    />
    <div
      className="right"
    />
  </div>
  <div
    className="timer"
  >
    Time Remaining: 
    5
  </div>
</div>
`;
